ui/selectors/selectors.js

```diff
@@ -495,8 +495,24 @@ export function getTargetAccount(state, targetAddress) {
   return accounts[targetAddress];
 }
 
-export const getTokenExchangeRates = (state) =>
-  state.metamask.contractExchangeRates;
+export const getTokenExchangeRates = (state) => {
+  const chainId = getCurrentChainId(state);
+  console.log('Here ....', state.metamask);
+  const contractMarketData = state.metamask.marketData?.[chainId] ?? {};
+
+  return Object.entries(contractMarketData).reduce(
```

Comment:
This selector may need to be memoized with `createDeepEqualSelector` or similar so things don't re-render unnecessarily.

I think this is because `useSelector` uses `===` for equality, and `Object.entries` is going to create a new object each time.